lib/widget/carousel.dart

```diff
@@ -241,8 +242,8 @@ class CarouselState extends WidgetState<Carousel> with TemplatedWidgetState {
     // wrap each child inside Container to add padding and gap
     double gap =
         widget._controller.gap?.toDouble() ?? MyController.defaultItemGap;
-    double leadingGap = widget._controller.leadingGap?.toDouble() ?? gap / 2;
-    double trailingGap = widget._controller.trailingGap?.toDouble() ?? gap / 2;
+    double leadingGap = widget._controller.leadingGap?.toDouble() ?? 0;
```

Comment:
So the reason leadingGap = gap/2 is because we want to scroll items from the screen edge, so margin is often time not effective. When margin is not used, user have to set both the leadingGap and trailingGap to be exactly the same for it to look right.

I think we are OK to set this to 0, but it just won't look good out of the box (unless you really want zero margin). Go ahead and merge this in.

Reply:
Okay I’ll merge this PR